### ISSUE TITLE
Advance logic for `--base` option

### ISSUE THREAD

**Issue by skywinder** [skywinder/github-changelog-generator#269]
It's addition for #258.

For now `--base` option just concat old change log with new, auto-generated. 
It would be nice to use full power of `Reader`, to not just fetch last tag, but also fetch `content` for each issue and put auto-generated section for each tag, and then data from `content`.

For example, for https://github.com/troessner/reek project

For tag `3.3.0`:

Form manual change log:

>## 3.3.0 (2015-08-23)
* (HParker / Adam Hess) add support for or-assign constants
* (troessner) Update TooManyMethods default value to 15

For auto-generated change log:

>## [v3.3.0](https://github.com/troessner/reek/tree/v3.3.0) (2015-08-22)
[Full Changelog](https://github.com/troessner/reek/compare/v3.2.1...v3.3.0)
**Fixed bugs:**
- Rake release doesn't work anymore with latest bundler [\#643](https://github.com/troessner/reek/issues/643)
- Rake task blows up with ‘Error: No such file - lib/\*\*/\*.rb’ [\#642](https://github.com/troessner/reek/issues/642)
- Make AppConfiguration's initializer flexible enough to deal with different inputs [\#608](https://github.com/troessner/reek/issues/608)
**Closed issues:**
- private method `load' called for Psych:Module [\#653](https://github.com/troessner/reek/issues/653)
**Merged pull requests:**
- Remove spec/gem directory. [\#659](https://github.com/troessner/reek/pull/659) ([troessner](https://github.com/troessner))
- Ensure each file can be individually required [\#658](https://github.com/troessner/reek/pull/658) ([mvz](https://github.com/mvz))
- etc...

---
In the end, I want to get something like this as output:

## [v3.3.0](https://github.com/troessner/reek/tree/v3.3.0) (2015-08-22)
[Full Changelog](https://github.com/troessner/reek/compare/v3.2.1...v3.3.0)

* (HParker / Adam Hess) add support for or-assign constants
* (troessner) Update TooManyMethods default value to 15

`[//]: # (START OF AUTO CHANGE LOG CONTENT)`

**Fixed bugs:**

- Rake release doesn't work anymore with latest bundler [\#643](https://github.com/troessner/reek/issues/643)
- Rake task blows up with ‘Error: No such file - lib/\*\*/\*.rb’ [\#642](https://github.com/troessner/reek/issues/642)
- Make AppConfiguration's initializer flexible enough to deal with different inputs [\#608](https://github.com/troessner/reek/issues/608)

**Closed issues:**

- private method `load' called for Psych:Module [\#653](https://github.com/troessner/reek/issues/653)

**Merged pull requests:**

- Remove spec/gem directory. [\#659](https://github.com/troessner/reek/pull/659) ([troessner](https://github.com/troessner))
- Ensure each file can be individually required [\#658](https://github.com/troessner/reek/pull/658) ([mvz](https://github.com/mvz))
- etc...

The string `[//]: # (START OF AUTO CHANGE LOG CONTENT)` will be invisible in markdown file, but helps to detect where manual change log ends (could be helpful in the future).

**Comment by raphink** [comment]
So you would use a merge approach. That could be really cool, as it would allow to store all manual entries aside, with just the tag for each, and let the automated part populate from GitHub + date.

**Comment by skywinder** [comment]
Yep. And your PR #258 was a good movement in that direction. I will handle it soon.

**Comment by fabn** [comment]
I've just discovered this project and already love it.

This issue (and possibly #56) is the number one issue in my watchlist.

Not every notable change is documented in GH issues so have a way to add manual entries to changelog file imho is a must.

That said thanks for this awesome gem.
